src/features/hoverProvider.ts

```diff
@@ -21,9 +21,34 @@ export default class OmniSharpHoverProvider extends AbstractSupport implements H
 
         return serverUtils.typeLookup(this._server, req, token).then(value => {
             if (value && value.Type) {
-                let contents = [extractSummaryText(value.Documentation), { language: 'csharp', value: value.Type }];
+                let structDoc = value.StructuredDocumentation ;
+                let newline = "\n\n";
+                let documentation = "";
+                Object.getOwnPropertyNames(structDoc).forEach(
+                    function (val) {
+                        if(Array.isArray(structDoc[val])){
+                            if(structDoc[val].length>0){
+                                if(val=="ParamElements"){
+                                    documentation += "Parameters:" ;
+                                }
+                                else if(val=="TypeParamElements"){
+                                    documentation += "TypeParameters:";
+                                }
+                                else{
+                                    documentation += "Exceptions:";
+                                }
+                                documentation += newline + structDoc[val].join(newline) + newline;
+                             }
+                         }
+                        else if(structDoc[val]){
+                            documentation += structDoc[val] + newline;
+                        } 
+                    }
+                );
```

Comment:
I'm not sure I'm sold on the property loop.  This will cause the order of the tooltip to change if the properties are ever returned in a different order. Also, if any other element that becomes an array would get the heading "Exceptions:". Should we just check the properties we care about and construct the tooltip text in a deterministic way? E.g.

```TypeScript
if (structDoc.SummaryText) {
    documentation += structDoc.SummaryText;
}

if (structDoc.ParamElements && structDoc.ParamElements.length > 0)) {
    documentation += "Parameters:" + newLine;
    documentation += structDoc.ParamElements.join(newLine) + newLine;
}

etc...
```